os/board/imxrt1050-evk/userspace/imxrt_userspace.c

```diff
@@ -0,0 +1,133 @@
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdlib.h>
+
+#include <tinyara/userspace.h>
+#include <tinyara/wqueue.h>
+#include <tinyara/mm/mm.h>
+
+#if defined(CONFIG_BUILD_PROTECTED) && !defined(__KERNEL__)
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+/* Configuration ************************************************************/
+
+#ifndef CONFIG_TINYARA_USERSPACE
+#error "CONFIG_TINYARA_USERSPACE not defined"
+#endif
+
+#if CONFIG_TINYARA_USERSPACE != 0x60200000
+#error "CONFIG_TINYARA_USERSPACE must be 0x60200000 to match user-space.ld"
+#endif
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/* These 'addresses' of these values are setup by the linker script.  They are
+ * not actual uint32_t storage locations! They are only used meaningfully in the
+ * following way:
+ *
+ *  - The linker script defines, for example, the symbol_sdata.
+ *  - The declareion extern uint32_t _sdata; makes C happy.  C will believe
+ *    that the value _sdata is the address of a uint32_t variable _data (it is
+ *    not!).
+ *  - We can recoved the linker value then by simply taking the address of
+ *    of _data.  like:  uint32_t *pdata = &_sdata;
+ */
+
+extern uint32_t _stext;			/* Start of .text */
+extern uint32_t _etext;			/* End_1 of .text + .rodata */
+extern const uint32_t _eronly;	/* End+1 of read only section (.text + .rodata) */
+extern uint32_t _sdata;			/* Start of .data */
+extern uint32_t _edata;			/* End+1 of .data */
+extern uint32_t _sbss;			/* Start of .bss */
+extern uint32_t _ebss;			/* End+1 of .bss */
+
+/* This is the user space entry point */
+
+int CONFIG_USER_ENTRYPOINT(int argc, char *argv[]);
+
+const struct userspace_s userspace __attribute__((section(".userspace"))) = {
+	/* General memory map */
+
+	.us_entrypoint = (main_t) CONFIG_USER_ENTRYPOINT, .us_textstart = (uintptr_t) &_stext, .us_textend = (uintptr_t) &_etext, .us_datasource = (uintptr_t) &_eronly, .us_datastart = (uintptr_t) &_sdata, .us_dataend = (uintptr_t) &_edata, .us_bssstart = (uintptr_t) &_sbss, .us_bssend = (uintptr_t) &_ebss,
```

Comment:
Remove the space between type casting and variable

Reply:
I will fix it